api/src/context.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_types::{chain_id::ChainId, ledger_info::LedgerInfoWithSignatures};
+use storage_interface::MoveDbReader;
+
+use anyhow::Result;
+use std::{borrow::Borrow, sync::Arc};
+
+// Context holds application scope context
+pub struct Context {
+    chain_id: ChainId,
+    db: Arc<dyn MoveDbReader>,
+}
```

Comment:
Given you plan on wrapping Context in an Arc anyways maybe we can consider changing the db to be a generic parameter instead of a trait object?

Reply:
Specifically something like this: https://github.com/bmwill/diem/commit/360a67246cfe850496af5439147e3fc528c01e5a

Reply:
removed Arc wrapping for Context, kept Arc<dyn MoveDbReader> as generic parameter version is more complex than expected.